### ISSUE TITLE
Exception in Xaml.MapControl.StartZoomAnimation

### ISSUE THREAD

**Issue by grudolf** [pauldendulk/Mapsui#97]
Hello,

using mousewheel to zoom on a map with a very small Viewport/BoundingBox (Height and Width are both 0 because I somehow managed to zoom the map to a single point) causes an ArgumentException:

```
System.ArgumentException: 'NaN' is not a valid value for property 'From'.
   at System.Windows.DependencyObject.SetValueCommon(DependencyProperty dp, Object value, PropertyMetadata metadata, Boolean coerceWithDeferredReference, Boolean coerceWithCurrentValue, OperationType operationType, Boolean isInternal)
   at System.Windows.DependencyObject.SetValueInternal(DependencyProperty dp, Object value)
   at Mapsui.UI.Xaml.MapControl.StartZoomAnimation(Double begin, Double end)
   at Mapsui.UI.Xaml.MapControl.MapControlMouseWheel(Object sender, MouseWheelEventArgs e)
   at System.Windows.Input.MouseWheelEventArgs.InvokeEventHandler(Delegate genericHandler, Object genericTarget)
   ...
```

Tested on Mapsui 0.8.0, current version source is identical.

**Comment by pauldendulk** [comment]
okay, thanks I look at it. I assume it is WPF?

**Comment by grudolf** [comment]
WPF, yes, here's a short sample:


MainWindow.xaml:

    <Window x:Class="MapsuiTest.MainWindow"
            xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
            xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
            xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
            xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
            xmlns:xaml="clr-namespace:Mapsui.UI.Xaml;assembly=Mapsui.UI.Xaml"
            mc:Ignorable="d"
            Title="MainWindow" Height="350" Width="525">
        <Grid>
            <Grid.RowDefinitions>
                <RowDefinition Height="Auto"></RowDefinition>
                <RowDefinition></RowDefinition>
            </Grid.RowDefinitions>
            <ToolBarTray VerticalAlignment="Top" Height="30" Grid.Row="0">
                <ToolBar VerticalAlignment="Top" Height="30">
                    <Button Height="25" Content="Zoom" Click="Test"></Button>
                </ToolBar>
            </ToolBarTray>

            <xaml:MapControl x:Name="MapControl" VerticalAlignment="Stretch" HorizontalAlignment="Stretch" Grid.Row="1"></xaml:MapControl>
        </Grid>
    </Window>



MainWindow.xaml.cs:

    using System.Windows;
    using BruTile.Predefined;
    using Mapsui.Layers;
    using Mapsui.Providers;
    using Point = Mapsui.Geometries.Point;

    namespace MapsuiTest
    {
        public partial class MainWindow : Window
        {
            private MemoryLayer _memoryLayer;

            public MainWindow()
            {
                InitializeComponent();

                MapControl.Map.Layers.Clear();
                MapControl.Map.Layers.Add(new TileLayer(KnownTileSources.Create()));
                var memoryProvider = new MemoryProvider();
                memoryProvider.Features.Add(new Feature { Geometry = new Point(0, 0) });
                _memoryLayer = new MemoryLayer
                {
                    Name = "TestLayer",
                    DataSource = memoryProvider
                };
                MapControl.Map.Layers.Add(_memoryLayer);
                MapControl.ZoomToFullEnvelope();
                MapControl.Refresh();
            }

            private void Test(object sender, RoutedEventArgs e)
            {
                var bbox = _memoryLayer.Envelope;
                MapControl.Map.NavigateTo(bbox);
            }
        }
    }

**Comment by pauldendulk** [comment]
I tried to reproduce it but get no exception. I used a Map sample based on your above:

```
    Map CreateTestMap()
    {
        var map = new Map();

        var memoryProvider = new MemoryProvider();
        memoryProvider.Features.Add(new Feature { Geometry = new Mapsui.Geometries.Point(0, 0) });
        map.Layers.Add(new MemoryLayer {DataSource = memoryProvider});

        return map;
    }
```
The resolution goes to zero but no exception. I assume this is fixed because of work that I did related to the empty map sample. 

Now, apart from no exception, you also would like to be able to restrict the zoom. This is the case whenever you use a tile layer. The ZoomHelper.ZoomIn/Out use the TileLayer resolutions. In cases where  there is no TileLayer the current version has no way to restricted zoom. This could be added by adopting some of the MIT code [here](https://github.com/itinero/ui/blob/master/src/Itinero.ui.Core/RestrictPanZoom.cs). This is not planned before 1.0.

**Comment by blwhttngtn** [comment]
(Mapsui1.48)

Hi @pauldendulk and team, I came across this today on a production build. I'm attaching an StackTrace screenshot.

On a testing machine, we determined that this issue is reproducible when, at the System Settings level, you swap the **Decimal Symbol** period ('.') and **Digit Grouping Symbol** comma ('.')

For us, we were crashing while navigating to a bounding box on startup.

![image](https://user-images.githubusercontent.com/12670350/64300244-f9edc580-cf30-11e9-9e97-57021dac1d02.png)

![image](https://user-images.githubusercontent.com/12670350/64299849-87c8b100-cf2f-11e9-9cae-570708e7f965.png)

![image](https://user-images.githubusercontent.com/12670350/64299686-f5c0a880-cf2e-11e9-913f-604560b5b1c4.png)
